sched/wqueue/kwork_notifier.c

```diff
@@ -256,6 +250,10 @@ int work_notifier_setup(FAR struct work_notifier_s *info)
 
       memcpy(&notifier->info, info, sizeof(struct work_notifier_s));
 
+      /* Disable interrupts very briefly. */
+
+      flags = enter_critical_section();
+
       /* Generate a unique key for this notification */
 
       notifier->key = work_notifier_key();
```

Comment:
Can this be moved out os the CS?

https://github.com/apache/incubator-nuttx/blob/cdd111a1faec9b40b707797e00c4afae4956fb3f/sched/wqueue/kwork_notifier.c#L146
Has a loop. 

How long are interrupts off?

Reply:
Hi @davids5 

> How long are interrupts off?
 it is  depends on the list size of "g_notifier_pinding", it seems good since there is just traverse of the list

Reply:
can we switch to a free running counter?

Reply:
> it is depends on the list size of "g_notifier_pinding", it seems good since there is just traverse of the list

That adds latency and jitters.

Let do as @xiaoxiang781216  suggest and change to free running counter.

Reply:
Done